src/main/java/com/exactpro/th2/FixHandler.java

```diff
@@ -504,12 +591,14 @@ private void reset() {
     }
 
     public void sendResendRequest(int beginSeqNo, int endSeqNo) { //do private
+        LOGGER.info("Sending resend request: {} - {}", beginSeqNo, endSeqNo);
         StringBuilder resendRequest = new StringBuilder();
         setHeader(resendRequest, MSG_TYPE_RESEND_REQUEST, msgSeqNum.incrementAndGet());
-        resendRequest.append(BEGIN_SEQ_NO).append(beginSeqNo).append(SOH);
-        resendRequest.append(END_SEQ_NO).append(endSeqNo).append(SOH);
+        resendRequest.append(BEGIN_SEQ_NO).append(beginSeqNo);
+        resendRequest.append(END_SEQ_NO).append(endSeqNo);
         setChecksumAndBodyLength(resendRequest);
-        channel.send(Unpooled.wrappedBuffer(resendRequest.toString().getBytes(StandardCharsets.UTF_8)), Collections.emptyMap(), null, SendMode.HANDLE_AND_MANGLE);
+        channel.send(Unpooled.wrappedBuffer(resendRequest.toString().getBytes(StandardCharsets.UTF_8)), Collections.emptyMap(), null, SendMode.HANDLE_AND_MANGLE)
+            .thenAcceptAsync(x -> strategy.getState().addMessageID(x));
```

Comment:
When we use `*Async` without executor a default executor will be used (either ForkJoinPool or ThreadPerTask depending on a number of available cores). In most cases in our clusters, it will be ThreadPerTask impl. If it is okay for us we can keep it. But probably, we should pass executor explicitly